lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -388,6 +394,33 @@ def configure(conf)
       @resource.labels.freeze
       @common_labels.freeze
 
+      # If prometheus monitoring is enabled, register metrics in the default
+      # registry and store metric objects for future use. If metrics are
+      # registed already, just store the metric objects.
+      if @prometheus_monitoring_enabled
+        require 'prometheus/client'
+
+        prometheus = Prometheus::Client.registry
+
+        # Exception-driven behavior to avoid synchronization errors
+        begin
+          @prometheus_http_requests_count = prometheus.counter(
```

Comment:
Oh, I've just realized that the implicit mechanism is much harder to test, since you have a dependency on the environment and you cannot test non-prometheus pipeline if prometheus gem is installed and vice versa (at least to my limited knowledge in ruby). Maybe `monitoring_type` is not such a bad idea?